### ISSUE TITLE
getLeagueBySummoner() - getEntries issue

### ISSUE THREAD

**Issue by vBandiTv** [rithms/riot-api-java#66]
Hi again!

I'm trying to get the summoner's league division, but i am getting many version of the division.
When i run the request on Riot's Web `https://euw.api.pvp.net/api/lol/euw/v2.5/league/by-summoner/62128873, 19881488/entry?api_key=<API-KEY>` 
I am getting the following json:
`{"62128873": [{
   "queue": "RANKED_SOLO_5x5",
   "name": "Kennen's Pyromancers",
   "entries": [{
      "leaguePoints": 31,
      "isFreshBlood": false,
      "isHotStreak": false,
      "division": "V",
      "isInactive": false,
      "isVeteran": false,
      "losses": 63,
      "playerOrTeamName": "vBandiTv",
      "playerOrTeamId": "62128873",
      "wins": 40
   }],
   "tier": "BRONZE"
}]}`

The devision in the above json is right, but when i try to get the division from Java, I am using the following method:
`String lDivision = api.getLeagueBySummoner(Region.EUW, api.getSummonerByName(sName).getId()).get(0).getEntries().get(0).getDivision()`

The problem is that this method returns a wrong division, and when i run .size() on the .getEntries() i get a size of 100, but on the json above the "entries" list have only 10 objects.

Is that some problem with the method? Or am I doing something wrong?

Thx in advance again!

**Comment by rithms** [comment]
Hi,

Please don't include your API key in our examples. I went ahead and edited your post to remove it.
Looking into your issue now. Just out of curiosity, which version of the library are you currently using?

**Comment by vBandiTv** [comment]
First of all thank you and sorry for the key things, i didn't even noticed that the url I'v copied has my key, thank you very much, i appreciate that!

The version of current library i have is 3.9.0.

**Comment by rithms** [comment]
Hi vBandiTv,

The Riot Games API has two different League endpoints.

One to get **the entire league of a summoner** (includes entries of the other 99 summoners in that same league): 

https://euw.api.pvp.net/api/lol/euw/v2.5/league/by-summoner/62128873,%2019881488?api_key=<API-KEY-GOES-HERE>

The library method that uses this endpoint is **getLeagueBySummoners()**.

and one just to get **the league entry of a summoner**, (only includes the summoners specified): 

https://euw.api.pvp.net/api/lol/euw/v2.5/league/by-summoner/62128873,%2019881488/entry?api_key=<API-KEY-GOES-HERE>

The library method that uses this endpoint is **getLeagueEntryBySummoners()**.


**TLDR: Based on the URL given in your example, I believe you want to use the method getLeagueEntryBySummoners() rather than getLeagueBySummoners().**

Hope this helps.

**Comment by vBandiTv** [comment]
Thx again rithms, that was the correct method for me!

One more question (I dont want to create a new issue).

The method getTopChampionMasteries() have an enum PlatformId.
What is the difference between the Enum PlatformId and the Enum Region?

**Comment by vBandiTv** [comment]
Ye, i saw that this 2 enums are the same, but the explanation about the end points made it clear why.

Anyway I'v found some way to convert them and it works, and if it will be changed on the future it still can be changed.

Thx again rithms!
